Add .gitignore content in advanced section (#243)
githubtraining/training-manual: docs/23_gitignore.md

## Configure Git to ignore files

> You can add a text file in your repository titled `.gitignore` that instructs Git to stop tracking changes of certain files. This also means that these files will not be pushed to GitHub from your local machine.

Reviewer: ```suggestion
While working within your repository, you might notice files that your preferred editor or operating system have place in the repository. These files are typically things that you don't want git to track or include in the history of your project. Although you can manually ignore those files by never committing them to the history of the project, there is an easier way for you to ignore those files. Git enables you to ignore specific files, files types, and folders through the use of a text file you can add to your  repository titled `.gitignore`. This file instructs Git to stop tracking changes of certain files, which means that these files will not be pushed to GitHub from your local machine.
```

**Files that you might want Git to ignore:**
1. Secrets, e.g. `*.pem`, `*.key`
1. Binary files, e.g. `*.pdf*`, `*.docx`
1. Log files, e.g. `*.log`
1. Build files, e.g. `Build/*`

### Here are some ways you can add a `.gitignore` file:

#### Ignore files  locally

In your Terminal or Git Bash, `cd` to the repository of interest and create a `.gitignore` file
```

> touch .gitignore

Reviewer: In general can we try to be more consistent when presenting the reader with steps to complete a task? In this specific instance, the user has created a `.gitignore` file, but hasn't committed it, added anything to the `.gitignore` file, etc. 

Ideally we would have some exercise that lets us show them how the `.gitignore` file works both from a "I can see how I could potentially use this" perspective and a "Oh, I can see it in action through this activity" perspective. 

Overall, great addition for this topic though 🎉

Reviewer 2: This might be useful here: https://gist.github.com/octocat/9257657

> While working within your repository, you might notice files that your preferred editor or operating system have place in the repository. These files are typically things that you don't want git to track or include in the history of your project. Although you can manually ignore those files by never committing them to the history of the project, there is an easier way for you to ignore those files. Git enables you to ignore specific files, files types, and folders through the use of a text file you can add to your  repository titled `.gitignore`. This file instructs Git to stop tracking changes of certain files, which means that these files will not be pushed to GitHub from your local machine.

Reviewer 2: ```suggestion
While working within your repository, you might notice system or configuration files created by your operating system or text editor. These are typically files that you don't want Git to track or include in the history of your project. You can configure Git to ignore specific files, files types, and folders by defining them in a text file titled `.gitignore` in your repository's root directory. 
```

> 1. Build files, e.g. `Build/*`

Reviewer 2: Maybe this section should be moved down below the steps to create an empty `.gitignore` file.

> #### Ignore files  locally

Reviewer 2: ```suggestion
#### Ignore files for a single repository
```

```

> 

Reviewer 2: ```suggestion

To share the ignore rules with other users who clone the repository, commit the `.gitignore` file in to your repository.

```

#### Ignore files for all repositories on your local machine

You can also create a global `.gitignore` file to define a list of rules for ignoring files in every local Git repository

```
git config --global core.excludesfile ~/.gitignore_global
```

#### github/gitignore

The [github/gitignore repository](https://github.com/github/gitignore) is a collection of `.gitignore` templates based on popular programming languages.

For example, the following snippet shows some suggested files you should start ignoring in a Python project.

![python-gitignore](img/python-gitignore.png ':size=60%')

These templates are also available when you create a new repository:

![create-gitignore-repo](img/gitignore-create-repo.png ':size=60%')

#### gitignore.io

[This website](https://www.toptal.com/developers/gitignore) is another resource for you to find appropriate `.gitignore` templates when your project contains multiple programming languages.

> ![gitignore.io](img/gitignore.io.png)

Reviewer: I don't know if we need this image necessarily, while the other images provide context regarding the use of GitHub provided `.gitignore` solutions, we don't have much insight as to any upcoming changes to this site so we might not be aware of any UI changes coming down the pipe. Including / Excluding this image from the file isn't a blocker, just my $.02

## Configure Git to ignore files

While working within your repository, you might notice files that your preferred editor or operating system have place in the repository. These files are typically things that you don't want git to track or include in the history of your project. Although you can manually ignore those files by never committing them to the history of the project, there is an easier way for you to ignore those files. Git enables you to ignore specific files, files types, and folders through the use of a text file you can add to your  repository titled `.gitignore`. This file instructs Git to stop tracking changes of certain files, which means that these files will not be pushed to GitHub from your local machine.

**Files that you might want Git to ignore:**

1. Secrets, e.g. `*.pem`, `*.key`

> 1. Binary files, e.g. `*.pdf*`, `*.docx`

Reviewer 2: ```suggestion
1. Binary files, e.g. `*.pdf`, `*.docx`
```

**Example files that you might want Git to ignore:**

1. Secrets, e.g. `*.pem`, `*.key`
1. Binary files, e.g. `*.pdf`, `*.docx`
1. Log files, e.g. `*.log`
1. Build files, e.g. `Build/*`

### Readily Available `.gitignore` Templates

#### github/gitignore

The [github/gitignore repository](https://github.com/github/gitignore) is a collection of `.gitignore` templates based on popular programming languages.

For example, the following snippet shows some suggested files you should start ignoring in a Python project.

![python-gitignore](img/python-gitignore.png ':size=60%')

These templates are also available when you create a new repository:

![create-gitignore-repo](img/gitignore-create-repo.png ':size=60%')

#### gitignore.io

[This website](https://www.toptal.com/developers/gitignore) is another resource for you to find appropriate `.gitignore` templates when your project contains multiple programming languages.

### Activity: Creating a `.gitignore` file

#### Ignore files for a single repository

1. In your Terminal or Git Bash, `cd` to a repository of interest and create a `.gitignore` file:

```sh

> touch .gitignore

Reviewer: `touch` doesn't work if the user is not using a bash-based terminal. Windows CMD users can use: `echo >> .gitignore` to accomplish the same thing.
